### DISCUSSION TITLE
Please remove telemetry

### DISCUSSION THREAD

**Discussion by ddevault** [meilisearch/product#162] [Feedback & Feature Proposal]
It is unethical to ship spyware to users, especially enabled by default. It is frustrating to see this. from a FOSS project. Please remove it from your software.

**Comment by gmourier** [comment]
Hello @ddevau!

We understand your position on this topic. 

We have put in place a process to communicate exhaustively on what is tracked and will be tracked in order to offer the best open-source search engine for our user's needs.
 
Here is the list of planned changes:

- We have added an explicit message at MeiliSearch launch giving quick access to the documentation to easily disable analytics. We merged the Sentry and Amplitude flags so that any user can easily deactivate our analytics tool at once. Here is the page we have specially edited on this topic https://dev.docs.meilisearch.com/learn/what_is_meilisearch/telemetry.html. This page will be updated each time we add data that we track.

- Each new specification lists the uses we want to track in the API for analytics purposes when there are any. Here's an example of the sort feature that should be released for version 0.22. https://github.com/meilisearch/specifications/blob/0b0b2e69ac412d04ee5bbdae673d5e62ef76bde7/text/0055-sort.md#i-measuring. 

Our goal is to be as transparent as possible about this because we think that's the best way to get people to understand why we need them today.

We don't track any sensitive or personally identifiable data using MeiliSearch. Also, we do not inspect or track data contained in indexes or documents.

The goal of all this is to provide a solution that meets your needs, that is our main objective.

I hope this answer will clarify any doubts you may have. We are aware that the term telemetry can be scary but the code is open and we will do our best to be totally transparent about it.

Thanks for your reading!

**Reply by ddevault** [reply]
This is not an acceptable answer. The use of opt-out telemetry is ethically wrong, and that speaks poorly for your integrity as a FOSS project, commercial product, and as a business. If nothing else, make it opt-in. But you should simply remove it.

**Reply by ddevault** [reply]
Alternatively, if you're willing to refer to it in your documentation as spyware, rather than euphemistically as telemetry, I would withdraw my objection.
